docs/source/whitepaper.rst

```diff
@@ -450,3 +450,36 @@ one additional week, which would result in higher opportunity costs for the liqu
 
 In practice the longest observed delay of block inclusion from a rollup sequencer has been 18 hours, and was exceptional.
 Hence the decision not to take this delay into account.
+
+Potential attacks
+-----------------
+
+Exhausting the agents funds
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Upon seeing a claim to a non-filled request, honest agents will challenge them. If they want to reduce their opportunity
+costs, they may challenge with a stake of ``claimStake + 1``. To which the dishonest claimer can respond with
+``2 * claimStake + 1``. After which the honest challenger will be tempted to stake enough to cover for the L1 non-fill
+proof. If he does so, the claimer can stop participating in the challenge and the honest challenger locked a higher
+stake than the claimer. This is a griefing attack on the opportunity costs of the challenger.
+
+The dishonest claimer can open parallel claims in an attempt to exhaust the funds of the honest agents. Once the honest
+agents have no funds, the dishonest participant is the sole participant of the protocol and can do as he pleases.
+
+For each opened non-filled claim, the attacker stakes ``claimStake - cost of L1 non-fill proof`` less than the honest
+agent. The advantage factor of the attacker is ``(claimStake - cost of L1 non-fill proof) / (2 * claimStake + 1)``. The
+attack is successful if
+
+::
+
+  ``total fund of attacker * advantage factor > total fund of honest agents``.
+
+The attacker will lose all it staked during the attack if liquidity providers discover the attack within the
+``challenge period`` and are able to refund their agents or manually trigger the L1 non-fill proof. However, for
+as long as it is the only participant, it will be able to wrongful claim any request and collect their rewards.
+
+A strategy could be put in place by the challenger to only ever outbid the claimer by 1. This would prevent such attack
+but it would take many more transactions to gather the funds for the L1 proof.
+
+Since the protocol is open and any participant can join with its funds, we believe for this attack to be unpractical and
+do not feel the need to mitigate it further.
```

Comment:
> I think I do not fully understand why this is only subject to non-filled requests...

I do not state that. There are various ways to grief the stake of the honest participants. I just wanted to make one explicit as a means of illustration.

> Since only one honest participant is needed this means that the attacker needs enough funds to potentially lock all funds of all participants which are potentially infinite.

In practice, an evildoer can assume that only liquidity providers participate in the protocol (at least at launch), they can check the addresses participating in the protocol and calculate their total funds.

> I don't see why this attack vector is not subject to filled requests as the attacker could do the same here. IIUC, the attack scenario results in a claim which nobody challenges because there are no funds left. Of course, a filled request has slightly different incentive structures, as the honest filler wants to receive back his tokens but those might be of very small value. If the claim of the honest filler is challenged an unchallenged claim might end sooner than all challenged claims and thus the attacker could withdraw the deposit.

I don't understand what you are talking about. It seems to me you believe that I state that once the attacker exhausted all the funds of the other participants, he can only attack the protocol by miss-using non-filled claims. Which is not what I write. Once the attacker is the honest participant, he can trivially steal all the funds locked in the contract by pretending having filled any request.

Reply:
> In practice, an evildoer can assume that only liquidity providers participate in the protocol (at least at launch), they can check the addresses participating in the protocol and calculate their total funds.

Good point. I agree.

> I don't understand what you are talking about. It seems to me you believe that I state that once the attacker exhausted all the funds of the other participants, he can only attack the protocol by miss-using non-filled claims. Which is not what I write. Once the attacker is the honest participant, he can trivially steal all the funds locked in the contract by pretending having filled any request.

That's not what I believe. But it's fine, you answered the point before by stating that you chose the Non-fill scenario as a means of illustration. So all good. And no I don't think that once all liquidity is locked, the attack is only possible to non-fills.

Reply:
> Whether you can go to L1 is irrelevant for this attack. I write "The attacker will lose all it staked during the attack if liquidity providers discover the attack within the challenge period and are able to refund their agents or manually trigger the L1 non-fill proof."
The reward of the attack being theoretically unbound, the attack might be worth the cost (I argue that it is not).

You are right.

Reply:
Maybe you could add in the beginning that the attack on a non-filled request is chosen as an example here.